test/Publish.ts

```diff
@@ -0,0 +1,1120 @@
+import { EmptyHomeScreen } from './helpers/EmptyHomeScreen';
+import { AccountsHome } from './helpers/flows/AccountsHome';
+import { App } from './helpers/flows/App';
+import { Network } from './helpers/flows/Network';
+import { HomeScreen } from './helpers/HomeScreen';
+import { CommonTransaction } from './helpers/messages/CommonTransaction';
+import { FinalTransactionScreen } from './helpers/messages/FinalTransactionScreen';
+import { OtherAccountsScreen } from './helpers/OtherAccountsScreen';
+import { Windows } from './helpers/Windows';
+import {
+  ISSUER_SEED,
+  USER_1_SEED,
+  USER_2_SEED,
+  WHITELIST,
+} from './utils/constants';
+import { faucet, getNetworkByte } from './utils/nodeInteraction';
+import {
+  ALIAS,
+  BURN,
+  CANCEL_LEASE,
+  DATA,
+  INVOKE_SCRIPT,
+  ISSUE,
+  LEASE,
+  MASS_TRANSFER,
+  REISSUE,
+  SET_ASSET_SCRIPT,
+  SET_SCRIPT,
+  SET_SCRIPT_COMPILED,
+  SPONSORSHIP,
+  TRANSFER,
+} from './utils/transactions';
+
+const WAVES = Math.pow(10, 8); // waves token scale
+type Account = { address: string; publicKey: string };
+
+describe('Publish', function () {
+  const hostNodeUrl = 'http://localhost:6869';
+  let chainId: number;
+  let issuer: Account, user1: Account, user2: Account;
+  let testAppTab: string;
+  let messageWindow: string | null = null;
+
+  let smartAssetId: string;
+  let assetWithMaxValuesId: string;
+
+  async function prepareAccounts() {
+    chainId = await getNetworkByte(hostNodeUrl);
+
+    issuer = {
+      address: address(ISSUER_SEED, chainId),
+      publicKey: publicKey(ISSUER_SEED),
+    };
+    user1 = {
+      address: address(USER_1_SEED, chainId),
+      publicKey: publicKey(USER_1_SEED),
+    };
+    user2 = {
+      address: address(USER_2_SEED, chainId),
+      publicKey: publicKey(USER_2_SEED),
+    };
+
+    await faucet({
+      recipient: issuer.address,
+      amount: 10 * WAVES,
+      nodeUrl: hostNodeUrl,
+    });
+  }
+
+  before(async function () {
+    await prepareAccounts();
+    await App.initVault();
+
+    const tabKeeper = await browser.getWindowHandle();
+    const { waitForNewWindows } = await Windows.captureNewWindows();
+    await EmptyHomeScreen.addButton.click();
+    const [tabAccounts] = await waitForNewWindows(1);
+
+    await browser.switchToWindow(tabKeeper);
+    await browser.closeWindow();
+
+    await browser.switchToWindow(tabAccounts);
+    await browser.refresh();
+
+    await Network.switchTo('Custom');
+    if (await CustomNetworkModal.root.isDisplayed()) {
+      await CustomNetworkModal.addressInput.setValue(this.nodeUrl);
+      await CustomNetworkModal.saveButton.click();
+    }
+
+    await AccountsHome.importAccount('user2', USER_2_SEED);
+    await AccountsHome.importAccount('user1', USER_1_SEED);
+    await AccountsHome.importAccount('issuer', ISSUER_SEED);
+
+    await Network.switchTo('Testnet');
+    await AccountsHome.importAccount('test', ISSUER_SEED);
+
+    await Network.switchTo('Custom');
```

Comment:
I cannot remove this b/c if i left Testnet network active transactions goes for Testnet account which has no funds :)

Reply:
I think he's saying that you can remove switching to testnet and back. Right, @smelnikov ? Because you don't seem to do anything in testnet anyway, why would you even switch to it?

Reply:
Removed